_aicspylibczi/Reader.cpp

```diff
@@ -199,9 +200,27 @@ namespace pylibczi {
               }
           }
       }
-      auto blk = m_czireader->ReadSubBlock(m_orderMapping.front().second);
-      auto info = blk->GetSubBlockInfo();
-      return info.logicalRect;
+
+      libCZI::IntRect lRect;
+      m_czireader->EnumerateSubBlocks([&lRect](int index, const libCZI::SubBlockInfo& info) -> bool {
+          if (!isPyramid0(info)) return true;
+
+          lRect = info.logicalRect;
+          return false;
+      });
+      return lRect;
+  }
+
+  libCZI::PixelType
+  Reader::getFirstPixelType(void)
+  {
+      libCZI::PixelType pixelType = libCZI::PixelType::Invalid;
+      m_czireader->EnumerateSubBlocks([&pixelType](int index_, const libCZI::SubBlockInfo& info_) -> bool {
+          if (!isPyramid0(info_)) return true;
+          pixelType = info_.pixelType;
+          return false;
+      });
+      return pixelType;
```

Comment:
Python convention would be to place this function above when it is used don't know about C though.

Reply:
class member functions can go anywhere